### PR TITLE
[ADD] step 4

### PR DESCRIPTION
**Pull request by lylich0**
Please kindly review the implementation of step 4.

### CODE REVIEW

File: mila/task_4/main.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+        # supposed sum of the values for the current char's new scaled hashmap
+        scaled_sum = round(total_sum / count) * given_length
+
+        for next_char in hashmap[char]:
+            if total_sum and scaled_sum:
+                hashmap[char][next_char] = round((hashmap[char][next_char] / total_sum) * scaled_sum)
+
+    while len(generated) != given_length:
+        next_probabilities = hashmap[current_char]
```
Comment (PavelDonchenko) [inline mila/task_4/main.py:65]: It's probably can be a chance that current_char key doesn't exist in hashmap

File: mila/task_4/main.py
Comment (PavelDonchenko) [inline mila/task_4/main.py:1]: I really tried  to contribute to your code quality,  but it was pretty hard with a lack of language knowledge) Actually, I chose the same approach for this step, so it looks good for me) P.S. Only can add, that this code needs some error handling, at least when the request is processed.

**Review by PavelDonchenko — APPROVED** [review]

File: mila/task_4/main.py (hunk 1 above)
Comment (lylich0) [reply mila/task_4/main.py:65]: "current_char" is obtained from the user's input. In the display() function, we validate user input by checking if the character is allowed, which guarantees it exists in the hashmap 👍

File: mila/task_4/main.py
Comment (lylich0) [reply mila/task_4/main.py:1]: Thank you! I have implemented error handling for the requests. Merging into the main branch.

**Pull request merged**